compiler/vm.nim

```diff
@@ -547,6 +547,23 @@ proc takeCharAddress(c: PCtx, src: PNode, index: BiggestInt, pc: int): TFullReg
   node.flags.incl nfIsPtr
   TFullReg(kind: rkNode, node: node)
 
+proc generateGenericAst(baseTyp: PNode): PNode =
+  ## Due to how generics work there is no existent AST, we need to generate it.
+  ## We can create approximate AST by replacing all generic parameter `ident`s with
+  ## the given type we have.
+  ## This does mean that generics with `when` inside do not appear properly from here.
+  result = copyTree(baseTyp[0].sym.ast)
```

Comment:
But why would there be a sym with a non-nil ast in there? That's only true for some tyGenericInst, not for all. For example: `type G[T] = T` doesn't have it iirc.

Reply:
Yea you are correct about that, uncertain how to best handle that. Seems there just is no consistent way to get AST for generics. Do you have any suggested paths to resolve this?

Reply:
I am not sure what problem you're solving here.

Reply:
`getImpl` presently does not work on generic types. This is a blocker for many macros such as `hasCustomPragma`.